packages/serve/src/index.ts

```diff
@@ -225,6 +225,16 @@ class ServeCommand {
             usedPorts.push(portNumber);
           }
 
+          // Set WEBPACK_HOT env variable to "true" if hot is specified in devServer options or via CLI
+          // If not specified, default to "true" as it is the default value for the hot option in devServer
+          if (typeof devServerOptions.hot !== "undefined") {
+            process.env.WEBPACK_HOT = String(
+              devServerOptions.hot === true || devServerOptions.hot === "only",
+            );
+          } else {
+            process.env.WEBPACK_HOT = "true";
```

Comment:
It should be early, so if you write `console.log(process.env["WEBPACK_HOT"]);` in `webpack.config.js`, it should be `true`. You need something this:
```js
process.env.WEBPACK_HOT = "true";

const compiler = await cli.createCompiler(webpackCLIOptions);

if (!compiler) {
  return;
}
```

But in other way you can disable `hot` inside `devServer` options, that's why I didn't implement this before, not sure how we can solve it.

We have `env` variables https://webpack.js.org/api/cli/#environment-options, and developer can create commands in npm `scripts` with hot and without hot, and get them in configuration or developer can just add them before `WEBPACK_HOT=true webpack serve`, I think we can't solve it the right way.

What do you think?

Reply:
@alexander-akait I think we can set it at two places. If we have passed `--hot`/`--no-hot` set it to `true`/`false` before loading the config. Then if users have overridden the `devServer.hot` in the config we can again override it after loading the config.

But yes, this behavior can be confusing to developers.

Reply:
> @alexander-akait I think we can set it at two places. If we have passed --hot/--no-hot set it to true/false before loading the config. Then if users have overridden the devServer.hot in the config we can again override it after loading the config.

We can, but it doesn't make sense, because you need `WEBPACK_HOT` in `webpack.config.js`, and it will be true in all cases except (`--no-hot`), so I don't see any possibles to solve it valid way, shorty - a developer want to get the value before we can read and evaluate it

And we really have `--env` variables for such purposes

Reply:
Yeah, `--env` makes more sense in this case.

Reply:
So let's close and provide examples, don't think we really need it